solvers/NonlinearProgram.m

```diff
@@ -155,7 +155,6 @@
       obj.bbcon_lb = [];
       obj.bbcon_ub = [];
       
-      obj = obj.setSolver('default');
```

Comment:
In a previous commit, I enclosed that line inside an ```if isempty(obj.solver)```, so that it wouldn't override the setting of the user (the user could eventually write ```obj.solver='something'``` somewhere). In the next commit I also plan to put a warning instead of an error and switch to fmincon.  Do you agree with these 2 changes?